### ISSUE TITLE
New easygo dependency breaks compile on some previously-working configurations

### ISSUE THREAD

**Issue by parnic** [mattermost/mattermost-server#17213]
#### Summary
The newly-added dependency on https://github.com/mailru/easygo (#16620) breaks compilation on some previously-working configurations.

#### Steps to reproduce
Compile Mattermost for an architecture such as `GOOS=netbsd GOARCH=arm`, e.g.:
```
cd cmd/mattermost
GOOS=netbsd GOARCH=arm go build
```

#### Expected behavior
Mattermost compiles successfully.

#### Observed behavior (that appears unintentional) 
Compilation fails. Depending on the platform/configuration, the error(s) are something like:
```
# github.com/mailru/easygo/netpoll
../../vendor/github.com/mailru/easygo/netpoll/kqueue.go:103:16: undefined: unix.EVFILT_USER
../../vendor/github.com/mailru/easygo/netpoll/kqueue.go:153:16: undefined: unix.EV_DISPATCH
../../vendor/github.com/mailru/easygo/netpoll/kqueue.go:164:15: undefined: unix.EV_RECEIPT
../../vendor/github.com/mailru/easygo/netpoll/kqueue.go:397:3: cannot use uint64(fd) (type uint64) as type uint32 in field value
../../vendor/github.com/mailru/easygo/netpoll/kqueue.go:398:3: cannot use int16(filter) (type int16) as type uint32 in field value
../../vendor/github.com/mailru/easygo/netpoll/kqueue.go:399:3: cannot use uint16(flags) (type uint16) as type uint32 in field value
```

For a larger list of now-failing architectures, see [SmartHoneybee's community-maintained build script](https://github.com/SmartHoneybee/ubiquitous-memory) that allows deploying of Mattermost to alternative architectures such as arm64 Raspberry Pis. [Here's an example](https://travis-ci.org/github/SmartHoneybee/ubiquitous-memory/builds/759379715) of a suite of passing builds for Mattermost 5.32.1, and [here's an example](https://travis-ci.org/github/SmartHoneybee/ubiquitous-memory/builds/763501851) of builds that started failing with v5.33.0.

#### Possible fixes
For some of the failures, using a fork of easygo that contains https://github.com/mailru/easygo/pull/17 could help, but there are other cases, such as the netbsd case, that are not covered by this PR. Presumably the best fix would be either maintaining a fork of easygo that works in more cases or replacing that dependency with another.

Note: I realize Mattermost doesn't officially support these configurations, but given that they worked up until 5.33.0, I figured this was the best opportunity to try and get it remedied rather than letting the problem languish. I understand if this is a won't-fix, but it's extremely helpful for those of us who run Mattermost on alternative platforms to have some of these more unusual configurations working.

**Comment by agnivade** [comment]
Thank you @parnic for the report. Let me look into it and figure out the best way to solve it.

**Comment by agnivade** [comment]
Hey @parnic - we discussed internally, and we decided to revert this feature altogether, as it was causing more pain than benefit. It should be available as a dot release in 5.33.2

**Comment by agnivade** [comment]
I am closing this as it's reverted in master. Dot release should be out soon. My apologies for the inconveniences @parnic, and thanks again for surfacing this.
